packages/@aws-cdk/aws-servicecatalogappregistry/lib/target-application.ts

```diff
@@ -91,9 +91,7 @@ class CreateTargetApplication extends TargetApplication {
     super();
   }
   public bind(scope: Construct): BindTargetApplicationResult {
-    const stackId = this.applicationOptions.stackId ?? 'ApplicationAssociatorStack';
-    (this.applicationOptions.stackName as string) =
-            this.applicationOptions.stackName || `Application-${this.applicationOptions.applicationName}-Stack`;
+    const stackId = this.applicationOptions.stackId ?? `Application-${this.applicationOptions.applicationName}-Stack`;
```

Comment:
`stackId` existed so you could change the name of the deployed stack, but that was the wrong mechanism, so we deprecated it in favor of the `stackName`. Now when you run `cdk list` or `cdk deploy` and you have to choose a stack id, you see a stack id that you didn't specify.

We don't need to re-introduce the `stackId` property to solve this; instead, we could set:
```suggestion
    (this.applicationOptions.stackName as string) =
            this.applicationOptions.stackName || `Application-${this.applicationOptions.applicationName}-Stack`;
    const stackId = this.applicationOptions.stackName;
```

Reply:
This makes sense, and it was an option the issue reporter was OK with. I pushed a new commit with your proposed change to assign stack id the value of stack name. Also updated the PR description to reflect this.